### PR TITLE
Filter party associations by survey id

### PR DESCRIPTION
**Pull request by jwgwalton**

### CODE REVIEW

File: ras_party/controllers/party_controller.py [hunk 1]
```diff
@@ -74,3 +74,30 @@ def get_party_by_id(sample_unit_type, id, session):
...
+            filtered_association['enrolments'] = filtered_enrolments
+            filtered_associations.append(filtered_association)
+
+    business['associations'] = filtered_associations
+
+    return business
+
+
+def filter_enrolments(existing_enrolments, survey_id):
+    filtered_enrolments = []
```
Comment (jcox-dev) [inline ras_party/controllers/party_controller.py:99]: 🎨 A list comprehension would be preferred here:

```python
return [enrolment for enrolment in existing_enrolments if enrolment['surveyId'] == survey_id]
```

File: ras_party/controllers/party_controller.py [hunk 2]
```diff
@@ -74,3 +74,30 @@ def get_party_by_id(sample_unit_type, id, session):
...
+
+
+def get_business_with_respondents_filtered_by_survey(sample_unit_type, id, survey_id):
+    business = get_party_by_id(sample_unit_type, id)
+
+    filtered_associations = []
+    for association in business['associations']:
+
+        filtered_association = {'partyId': association['partyId']}
+
```
Comment (jcox-dev) [inline ras_party/controllers/party_controller.py:86]: :art: Can probably remove this extra line.

**Review by jcox-dev — APPROVED** [review]
Couple of comments but otherwise works as expected.

File: ras_party/controllers/party_controller.py (hunk 1 above)
Comment (jwgwalton) [reply ras_party/controllers/party_controller.py:99]: I'm split on list comprehensions, sometime I think they make it less comprehendible. That one is pretty clear though. The end result of this babbling is that you're right and I will change it.

**Review by pricem14pc — APPROVED** [review]
Tested in PREPROD

**Pull request merged**